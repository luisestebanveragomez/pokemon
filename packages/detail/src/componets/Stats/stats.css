.stats__title {
  font-weight: 700;
  font-size: 2rem;
  color: var(--white);
  margin-bottom: 20px;
}