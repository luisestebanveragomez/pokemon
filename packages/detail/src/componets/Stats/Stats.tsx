import { FC } from "react";

import "./stats.css";
import { Pokemon } from "../../models/pokemonDetail.model";
import { Stat } from "../Stat/Stat";

interface StatsProps {
  detail: Pokemon;
  title: string;
  refStat: string;
  t: (key: string) => string;
}

export const Stats: FC<StatsProps> = ({ detail, t, refStat, title }) => {
  return (
    <>
      <h3 className="stats__title">{t(title)}</h3>
      {detail.stats.map(({ base, stat, percentage }) => (
        <Stat
          key={stat}
          base={base}
          percentage={percentage}
          name={t(`${refStat}${stat}`)}
        />
      ))}
    </>
  );
};
