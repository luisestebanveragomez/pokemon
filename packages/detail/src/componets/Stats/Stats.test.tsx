import { render } from '@testing-library/react';
import { Stats } from './Stats';

describe('Stats Component', () => {
  it('Should Be Render', () => {
    const { container } = render(<Stats detail={null} title={''} refStat={''} t={(el: string) => { return ''}} />)
    expect(container).toMatchSnapshot()
  })

  it('Should Be Render Values', () => {
    const data = {
      name: '',
      type: 'string',
      stats: [{base: 12,
        stat: '',
        percentage: 10}]
    }
    const { container } = render(<Stats detail={data} title={'test'} refStat={''} t={(el: string) =>  el} />)
    const title = document.querySelector('h3')
    expect(title.textContent).toEqual('test')
  })
})