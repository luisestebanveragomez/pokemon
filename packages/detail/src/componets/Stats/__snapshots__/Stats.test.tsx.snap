// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Component Should Be Render 1`] = `
<div>
  <h3
    class="stats__title"
  />
</div>
`;
